flagengine/flagengine_integration_test.go

```diff
@@ -3,61 +3,79 @@ package flagengine_test
 
 func TestEngine(t *testing.T) {
 	t.Parallel()
-	var testData struct {
-		Environment environments.EnvironmentModel `json:"environment"`
-		TestCases   []struct {
-			Identity identities.IdentityModel `json:"identity"`
-			Response struct {
-				Traits []traits.TraitModel          `json:"traits"`
-				Flags  []features.FeatureStateModel `json:"flags"`
-			} `json:"response"`
-		} `json:"identities_and_responses"`
-	}
 
-	testSpec, err := os.ReadFile(TestData)
+	// Read all test case files from the test_cases directory (both .json and .jsonc)
+	jsonFiles, err := filepath.Glob(filepath.Join(TestDataDir, "*.json"))
 	require.NoError(t, err)
-	require.NotEmpty(t, testSpec)
 
-	err = json.Unmarshal(testSpec, &testData)
+	jsoncFiles, err := filepath.Glob(filepath.Join(TestDataDir, "*.jsonc"))
 	require.NoError(t, err)
 
-	for i, c := range testData.TestCases {
-		t.Run(strconv.Itoa(i)+":"+c.Identity.CompositeKey(), func(t *testing.T) {
-			assert := assert.New(t)
-			require := require.New(t)
-			actual := flagengine.GetIdentityFeatureStates(&testData.Environment, &c.Identity)
-			expected := c.Response.Flags
-
-			sort.Slice(actual, func(i, j int) bool {
-				return actual[i].Feature.Name < actual[j].Feature.Name
-			})
-			sort.Slice(expected, func(i, j int) bool {
-				return expected[i].Feature.Name < expected[j].Feature.Name
-			})
-
-			require.Len(actual, len(expected))
-			for i := range expected {
-				id := strconv.Itoa(c.Identity.DjangoID)
-				assert.Equal(expected[i].Value(id), actual[i].Value(id))
-				assert.Equal(expected[i].Enabled, actual[i].Enabled)
+	files := append(jsonFiles, jsoncFiles...)
+	require.NotEmpty(t, files, "No test case files found in %s", TestDataDir)
+
+	for _, testFile := range files {
+		testFile := testFile // Capture range variable
+
+		// Get test name by removing extension
+		testName := filepath.Base(testFile)
+		testName = strings.TrimSuffix(testName, filepath.Ext(testName))
+
+		t.Run(testName, func(t *testing.T) {
+			t.Parallel()
+
+			// Read the test case file
+			testSpec, err := os.ReadFile(testFile)
+			require.NoError(t, err)
+			require.NotEmpty(t, testSpec)
+
+			// Standardise .jsonc files to standard JSON
+			if strings.HasSuffix(testFile, ".jsonc") {
+				ast, err := hujson.Parse(testSpec)
+				require.NoError(t, err)
+				ast.Standardize()
+				testSpec = ast.Pack()
+			}
+
+			// Parse the test case
+			var testCase struct {
+				Context engine_eval.EngineEvaluationContext `json:"context"`
+				Result  engine_eval.EvaluationResult        `json:"result"`
+			}
+
+			err = json.Unmarshal(testSpec, &testCase)
+			require.NoError(t, err)
+
+			// Run the evaluation
+			actual := flagengine.GetEvaluationResult(&testCase.Context)
+			expected := testCase.Result
+
+			// Compare the results
+			assert.Equal(t, expected.Flags, actual.Flags, "Flags should match")
+
+			// Compare segments - check key and name only since metadata is an implementation detail
```

Comment:
Is it possible to ensure the metadata in _result_ is equivalent to the metadata from `context`? This would assert the metadata is passed along by the engine.

Reply:
It is doing now